keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -1271,11 +1273,12 @@ type ResourceClaimStatus struct {
 	// +optional
 	DriverName string
 
-	// Allocation is set by the resource driver once a resource has been
-	// allocated successfully. If this is not specified, the resource is
-	// not yet allocated.
+	// Allocation is set by the resource driver once a (set of) resources has
+	// been allocated successfully. If this is not specified, the resources
+	// have not been allocated yet.
 	// +optional
-	Allocation *AllocationResult
+	// +listType=atomic
+	Allocation []AllocationResult
```

Comment:
I don't fully understand why do we want to have more than one `AllocationResult` object for a single allocation. Would it make sense to have a list of Kubelet Plugins as a field in the `AllocationResult`?

Reply:
Because each individual resource being allocated through a given kubelet plugin will only know how to process its own format for the RespurceHamdle (not a combined one). Moreover, depending on the semantics of the claim, each resource may want to be made available on a different set of nodes, thus requiring a different nodeSelector.

Reply:
I'm still not convinced. If ResourceHandle can be different for different Kubelet plugins we can try change this on the lower level and keep 1:1 relationship between allocation and allocation result.

Reply:
Because the claim might be backed by multiple different allocations that are completely independent.

Reply:
To me this PR looks good now, but I won't lgtm it if @bart0sh still has concerns.

Reply:
> I'm still not convinced. If ResourceHandle can be different for different Kubelet plugins we can try to change this on the lower level and keep 1:1 relationship between allocation and allocation result.

How would you propose doing this?

To me it is logical (as well as elegant) to have a list of `AllocationResults` here. It provides a succinct, obvious way for a controller to communicate which part of the overall claim allocation should be done by each kubelet plugin.

If I understand what you are suggesting correctly, you would (at a minimum) (1) list out the kubelet plugins inside the single allocation result, (2) impose some structure into the `ResourceHandle` so that the  kubelet can disaggregate the per-plugin data meant for each plugin.

Independent of that, I also foresee use-cases where it makes sense to service each "sub-allocation" on a different set of nodes with different sharing properties (which are the other two fields in the `AllocationResult` struct).

Reply:
yes, I thought about either extending `ResourceHandle` or wrapping `ResourceHandle` and `KubeletPluginName` into a separate struct.  However, considering "sub-allocation" on a different set of nodes and similar use cases it doesn't make much sense.